api-test/internal/conformance/client.go

```
@@ -0,0 +1,334 @@
+}
+
+// doAbs performs the request and records it (headers/cookies/bodies) into the
+// client's call trace for the debug report.
+func (c *Client) doAbs(label, method, absURL string, body []byte, contentType string) ([]byte, int, error) {
+	var rdr io.Reader
+	if body != nil {
+		rdr = bytes.NewReader(body)
+	}
+	// Seq is assigned up front so the error paths below record an ordered row too.
+	call := result.HTTPCall{Seq: len(c.calls) + 1, At: time.Now().UnixNano(), Label: label, Method: method, URL: absURL, ReqBody: string(body)}
+	req, err := http.NewRequest(method, absURL, rdr)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)